### ISSUE TITLE
Accessibility: `ReadOnlyField` is missing `role` attribute

### ISSUE THREAD

**Issue by GuySartorelli** [silverstripe/silverstripe-admin#1963]
[ARIA 1.2](https://www.w3.org/TR/wai-aria-1.2/#namefromprohibited) Cannot use aria-label or aria-labelledby on elements and roles that prohibit naming.

`ReadOnlyField` is represented by a `<p>` tag which has an implicit role `paragraph`. This role cannot be named/labelled.

We should add `role="input"` and `aria-readonly="true"` attributes to that element

## Notes

- We could alternatively use an actual `<input readonly>` element - though that will mean:
    - changing markup which may break some customisations
    - we can no longer use _(None)_ as a way to denote empty fields (they'll just be empty instead, which is probably better)

## Acceptance criteria

- [ ] `ReadOnlyField` markup has a role that allows naming and uses the `aria-readonly` attribute (or is replaced with a real readonly `input` element)
- [ ] Check both entwine and react forms

[Kitchen sink CI](https://github.com/creative-commoners/recipe-kitchen-sink/actions/runs/17280489579)

## PRs
- https://github.com/silverstripe/silverstripe-framework/pull/11844
- https://github.com/silverstripe/silverstripe-admin/pull/1991
- https://github.com/silverstripe/silverstripe-advancedworkflow/pull/609

**Comment by emteknetnz** [comment]
I've implemented as:

```html
<tag
  role="textbox"
  aria-readonly="true"
  tabindex="0"
>
```

aria-readonly is not supported by role="input" and will trigger the following js lint errors:

```
673:9   error  The attribute aria-readonly is not supported by the role input     jsx-a11y/role-supports-aria-props
675:11  error  Elements with ARIA roles must use a valid, non-abstract ARIA role  jsx-a11y/aria-role
```

So I've used textbox instead which does allow aria-readonly="true"

tabindex="0" is required to make this non-interactive element focusable. Without it, keyboard and screen reader users cannot navigate to the element.

I've had a good look through all fields in the codebase and have updated a few fields beyond just ReadOnlyField which uses non `<input>` tags when in a readonly state

**Comment by GuySartorelli** [comment]
PRs merged
